src/VectorTypes_Native.f90

```diff
@@ -363,5 +430,379 @@ SUBROUTINE getRange_RealVectorType(thisVector,istt,istp,getval,ierr)
       IF(PRESENT(ierr)) ierr=ierrc
     ENDSUBROUTINE getRange_RealVectorType
 
+#ifdef HAVE_MPI
+    SUBROUTINE init_NativeDistributedVectorType(thisVector,Params)
+      CHARACTER(LEN=*),PARAMETER :: myName='init_NativeDistributedVectorType'
+      CLASS(NativeDistributedVectorType),INTENT(INOUT) :: thisVector
+      TYPE(ParamType),INTENT(IN) :: Params
+      TYPE(ParamType) :: validParams
+      INTEGER(SIK) :: n, chunksize, comm, nProc,rank,mpierr,nlocal
+      INTEGER(SIK),ALLOCATABLE :: offsets(:)
+
+      !Check to set up required and optional param lists.
+      IF(.NOT.VectorType_Paramsflag) CALL VectorType_Declare_ValidParams()
+
+      !Validate against the reqParams and OptParams
+      validParams=Params
+      CALL validParams%validate(NativeDistributedVectorType_reqParams,NativeDistributedVectorType_optParams)
+
+      !Pull Data from Parameter List
+      CALL validParams%get('VectorType->n',n)
+      CALL validParams%get('VectorType->MPI_Comm_ID',comm)
+      CALL validParams%get('VectorType->nlocal',nlocal)
+
+      chunksize = 1
+      IF (validParams%has('VectorType->chunkSize')) THEN
+        CALL validParams%get('VectorType->chunkSize',chunkSize)
+      ENDIF
+
+      REQUIRE(.NOT. thisVector%isInit)
+      REQUIRE(n > 1)
+      REQUIRE(chunkSize > 0)
+      REQUIRE(MOD(n,chunkSize)==0)
+
+      thisVector%isInit=.TRUE.
+      thisVector%n=n
+      thisVector%chunkSize = chunkSize
+      thisVector%comm = comm
+
+      CALL MPI_Comm_rank(comm,rank,mpierr)
+      CALL MPI_Comm_size(comm,nproc,mpierr)
+
+      ! Default to greedy partitioning, respecting chunk size
+      IF (nlocal < 0) THEN
+        n = n/chunkSize
+        IF(rank < MOD(n,nproc)) THEN
+          thisVector%offset = (rank)*(n/nproc + 1)
+          thisVector%nlocal = n/nproc + 1
+        ELSE
+          thisVector%offset = (rank)*(n/nproc) + MOD(n,nproc)
+          thisVector%nlocal = n/nproc
+        ENDIF
+        thisVector%offset = thisVector%offset*chunkSize
+        thisVector%nlocal = thisVector%nlocal*chunkSize
+      ELSE
+        REQUIRE(MOD(nlocal,chunksize)==0)
+        thisVector%nlocal = nlocal
+        ALLOCATE(offsets(nproc))
+        CALL MPI_AllGather(nlocal,1,MPI_INTEGER,offsets(1),1,MPI_INTEGER,comm,mpierr)
+        thisVector%offset = SUM(offsets(1:rank))
+      ENDIF
+
+      ALLOCATE(thisVector%b(thisVector%nlocal))
```

Comment:
This should be initialized to something.

Reply:
New code allocates this vector as an offset array and sets to zero